java.source.base/src/org/netbeans/modules/java/source/base/SourceLevelUtils.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */package org.netbeans.modules.java.source.base;
```

Comment:
Formatting fixed. In the public API, there is javax.lang.model.SourceVersion and CompilationInfo.getSourceVersion(), usable for the clients. The only thing that it does not provide is the "allowFeature" method(s). I'd actually prefer to use the SourceVersion instead of the (internal) Source even in java.source.base, etc., but didn't experiment/work on that.